kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/filter/FilterAndInvoker.java

```diff
@@ -12,10 +12,11 @@
 
 /**
  * A Filter and it's respective invoker
- * @param filter filter
- * @param invoker invoker
+ * @param filterName The name of the filter (from the config)
+ * @param filter filter The filter instance
+ * @param invoker invoker The invoker
  */
-public record FilterAndInvoker(Filter filter, FilterInvoker invoker) {
+public record FilterAndInvoker(String filterName, Filter filter, FilterInvoker invoker) {
```

Comment:
I suspect the question is a can of worms, but why doesn't the name become a property of the Filter itself? Or do we not have a `NamedFilter` type we could pass around?

Reply:
`Filter` is the `interface` that filter developers implement ⇒ It can't a field.

I did consider adding a `NamedFilter`, but I think it would just have ended up in all the same places where we're currently passing a `FilterAndInvoker` (and not really any other places). So basically `FilterAndInvoker` (with this change) serves that same purpose.